### PR TITLE
[improvement] Include headers in response object

### PR DESCRIPTION
**Pull request by ellisjoe**
## After this PR
Headers are now accessible in the `Response` object.

### CODE REVIEW

File: dialogue-target/src/main/java/com/palantir/dialogue/Response.java [hunk 1]
```diff
@@ -26,6 +28,9 @@
     /** The HTTP response code for this response. */
     int code();
 
+    /** The HTTP headers for this response. */
+    Map<String, List<String>> headers();
```
Comment (carterkozak) [inline dialogue-target/src/main/java/com/palantir/dialogue/Response.java:34]: Does it make sense to provide both this and `Optional<String> contentType();`?

File: dialogue-java-client/src/main/java/com/palantir/dialogue/HttpChannel.java [hunk 2]
```diff
@@ -118,6 +119,11 @@ public int code() {
                 return response.statusCode();
             }
 
+            @Override
+            public Map<String, List<String>> headers() {
+                return response.headers().map();
```
Comment (markelliot) [inline dialogue-java-client/src/main/java/com/palantir/dialogue/HttpChannel.java:123]: idk ... is it worth a defensive copy here?

File: dialogue-target/src/main/java/com/palantir/dialogue/Response.java [hunk 3]
```diff
@@ -26,6 +28,9 @@
     /** The HTTP response code for this response. */
     int code();
 
+    /** The HTTP headers for this response. */
```
Comment (carterkozak) [inline dialogue-target/src/main/java/com/palantir/dialogue/Response.java]: Can we specify that this map uses a case-insensitive comparison?
https://tools.ietf.org/html/rfc7540#section-8.1.2

File: dialogue-target/src/main/java/com/palantir/dialogue/Response.java (hunk 1 above)
Comment (ellisjoe) [reply dialogue-target/src/main/java/com/palantir/dialogue/Response.java:34]: I replaced it with `Optional<String> getFirstHeader(String header)` which seems like the nice behavior you'd want

File: dialogue-java-client/src/main/java/com/palantir/dialogue/HttpChannel.java (hunk 2 above)
Comment (ellisjoe) [reply dialogue-java-client/src/main/java/com/palantir/dialogue/HttpChannel.java:123]: Can verify that this is an unmodifiable map: https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/master/src/java.net.http/share/classes/java/net/http/HttpHeaders.java#L117

File: dialogue-target/src/main/java/com/palantir/dialogue/Response.java (hunk 3 above)
Comment (ellisjoe) [reply dialogue-target/src/main/java/com/palantir/dialogue/Response.java]: done

**Review by markelliot — APPROVED** [review]

**Pull request merged**